pylib/Tools/Harasser/Harasser.py

```diff
@@ -0,0 +1,170 @@
+        return
+
+    def get_running_harassers(self):
+        return self.running_harassers
+
+    def parallel_execute(self, cmds, cmdargs, testDef):
+        status,stdout,stderr,time = testDef.execmd.execute(cmds, cmdargs, testDef)
+
+    def start(self, log, testDef):
+
+        trigger_scripts = self.options['trigger_scripts'][0]
+        if trigger_scripts is None or trigger_scripts == '':
+            trigger_scripts = []
+        else:
+            trigger_scripts = trigger_scripts.split(',')
+        stop_scripts = self.options['stop_scripts'][0]
+        if stop_scripts is None or stop_scripts == '':
+            stop_scripts = []
+        else:
+            stop_scripts = stop_scripts.split(',')
+
+        exec_ids = []
+        for trigger_script,stop_script in zip(trigger_scripts,stop_scripts):
+            trigger_script = trigger_script.strip()
+            stop_script = stop_script.strip()
+            ops = {(k[:-1] if k.endswith('_scripts') else k):\
+                   (trigger_script if k == 'trigger_scripts' else (\
+                    stop_script if k == 'stop_scripts' \
+                    else v[0])) \
+                   for k,v in self.options.items()}
+
+            cmdargs = trigger_script.split()
+
+            process = multiprocessing.Process(name='p'+str(self.execution_counter), \
+                            target=self.parallel_execute, \
+                            args=({k:v[0] for k,v in self.options.items()},cmdargs,testDef))
+            process.start()
+            
+            self.running_harassers[self.execution_counter] = (process, ops, datetime.datetime.now())
+            exec_ids.append(self.execution_counter)
+            self.execution_counter += 1
+
+        return exec_ids
+
+    def check(self, exec_ids, log, testDef):
+        dead_processes = []
+        for exec_id in exec_ids:
+            if not self.running_harassers[exec_id][0].is_alive() \
+                and self.running_harassers[exec_id][0].exitcode != 0:
+                dead_processes.append(self.running_harassers[exec_id])
+        if dead_processes:
+            process_run_info = self.stop(exec_ids, log, testDef)
+            return dead_processes, process_run_info
+        return None
+
+    def stop(self, exec_ids, log, testDef):
+
+        # TODO: stop all processes and return run info (status,stdout,stderr,time)
+        process_info = [self.running_harassers[exec_id] for exec_id in exec_ids]
+        for exec_id in exec_ids:
+            del self.running_harassers[exec_id]
+
+        return_info = []
+        for process,ops,starttime in process_info:
+            cmdargs = ops['stop_script'].split()
+            status,stdout,stderr,time = testDef.execmd.execute({k:v[0] for k,v in self.options.items()}, cmdargs, testDef)
+            return_info.append((status,stdout,stderr,datetime.datetime.now()-starttime))
+
+        for (process,ops,starttime),(status,stdout,stderr,time) in zip(process_info,return_info):
```

Comment:
Would you please add some comments explaining what is going on here? I am unfamiliar with this "zip" command, and I'm not sure I understand all the logic in this file